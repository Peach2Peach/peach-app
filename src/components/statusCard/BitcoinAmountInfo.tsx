import { memo } from "react";
import { View } from "react-native";
import tw from "../../styles/tailwind";
import i18n from "../../utils/i18n";
import { BTCAmount } from "../bitcoin/BTCAmount";
import { FixedHeightText } from "../text/FixedHeightText";
import { infoContainerStyle } from "./infoContainerStyle";

export const BitcoinAmountInfo = memo(
  ({ amount, premium }: { amount: number; premium?: number }) => (
    <View style={[infoContainerStyle, tw`gap-6px`]}>
      <BTCAmount size="small" amount={amount} />
      {premium !== undefined && (
        <FixedHeightText style={tw`body-m text-black-65`} height={17}>
          {Math.abs(premium)}%{" "}
          {i18n(
            premium > 0 ? "offer.summary.premium" : "offer.summary.discount",
          )}
        </FixedHeightText>
      )}
    </View>
  ),
);
