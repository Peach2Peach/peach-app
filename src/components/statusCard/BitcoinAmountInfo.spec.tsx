import { render } from "test-utils";
import { BitcoinAmountInfo } from "./BitcoinAmountInfo";

describe("BitcoinAmountInfo", () => {
  it("renders correctly", () => {
    const { toJSON } = render(
      <BitcoinAmountInfo
        {...{
          amount: 1000,
          premium: 1,
          chain: "bitcoin",
        }}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
