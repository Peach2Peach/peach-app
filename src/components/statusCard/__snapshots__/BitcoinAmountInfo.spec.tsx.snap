// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BitcoinAmountInfo renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "flex-end",
        "height": 40,
        "paddingBottom": 1,
        "paddingTop": 4,
        "width": 141,
      },
      {
        "gap": 6,
      },
    ]
  }
>
  <View
    style={
      [
        undefined,
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        {
          "height": 10,
          "width": 130,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "flexShrink": 0,
          },
          {
            "paddingBottom": 2,
            "paddingLeft": 2,
            "paddingRight": 2,
            "paddingTop": 2,
          },
        ]
      }
    >
      <test-file-stub
        fill="#888"
        style={
          [
            {
              "height": 12,
              "width": 12,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexBasis": "0%",
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "gap": 2,
          },
        ]
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexBasis": "0%",
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "color": "#2B1911",
                "fontFamily": "Baloo2-Regular",
                "fontSize": 16,
                "letterSpacing": 0.16,
                "lineHeight": 24,
              },
              [
                {
                  "opacity": 0.1,
                  "textAlign": "right",
                },
                [
                  {
                    "fontFamily": "Baloo2-SemiBold",
                    "fontSize": 14,
                    "letterSpacing": 0.14,
                    "lineHeight": 22,
                    "marginBottom": -10,
                    "marginTop": -10,
                  },
                  false,
                ],
              ],
              false,
            ]
          }
        >
          0.00 00
        </Text>
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "color": "#2B1911",
                "fontFamily": "Baloo2-Regular",
                "fontSize": 16,
                "letterSpacing": 0.16,
                "lineHeight": 24,
              },
              [
                {
                  "textAlign": "right",
                },
                [
                  {
                    "fontFamily": "Baloo2-SemiBold",
                    "fontSize": 14,
                    "letterSpacing": 0.14,
                    "lineHeight": 22,
                    "marginBottom": -10,
                    "marginTop": -10,
                  },
                  false,
                ],
              ],
              false,
            ]
          }
        >
          1 000
        </Text>
      </View>
      <Text
        allowFontScaling={false}
        style={
          [
            {
              "color": "#2B1911",
              "fontFamily": "Baloo2-Regular",
              "fontSize": 16,
              "letterSpacing": 0.16,
              "lineHeight": 24,
            },
            [
              {
                "fontFamily": "Baloo2-SemiBold",
                "fontSize": 14,
                "letterSpacing": 0.14,
                "lineHeight": 22,
                "marginBottom": -10,
                "marginTop": -10,
              },
              false,
            ],
            false,
          ]
        }
      >
        sats
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "height": 17,
        "justifyContent": "center",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        [
          {
            "color": "#2B1911",
            "fontFamily": "Baloo2-Regular",
            "fontSize": 16,
            "letterSpacing": 0.16,
            "lineHeight": 24,
          },
          [
            {
              "color": "#7D675E",
              "fontFamily": "Baloo2-Regular",
              "fontSize": 16,
              "letterSpacing": 0.16,
              "lineHeight": 24,
            },
            {
              "marginBottom": -40,
              "marginTop": -40,
            },
          ],
          false,
        ]
      }
    >
      1
      % 
      premium
    </Text>
  </View>
</View>
`;
